### ISSUE TITLE
create and secure a PAT for testing

### ISSUE THREAD

**Issue by jennybc** [jennybc/githug#9]
@gaborcsardi Are you willing to discuss this? Maybe it will apply to `gh` too. Now that I have a GitHub-API-calling function (`githug_init()`), I need to deal with having a PAT for testing. 😕

I don't really want repos being created and deleted in my own account. I think I should create a new user just for this. And then get a PAT with the proper scopes. I already know how to use a secure env var on Travis. The bigger question is how to set things up on my own machine.

When I'm working on the package, I want to use my normal `GITHUB_PAT`, linked by GitHub account. Only when running tests via `testthat` and the related `devtools` wrappers do I want to use this other PAT. I know to use `Sys.setenv()` to override env vars for a given R session, so I could put that in a helper script, with the dummy account's PAT. Then I would ... gitignore that helper? So it's sourced prior to local test runs but never even goes to GitHub and therefore Travis. Does that sound sane?

**Comment by gaborcsardi** [comment]
Yes, I think the helper is a good idea. It is somewhat simpler to put only the env var in a file, and ignore that, and then the helper is in the repo. The less is ignored the better, ignored things tend to get lost.....

Then in `testthat.R` you can just get it from the ignored file (it can be in the same directory) and then say `with_envvar(test_package, ...)` This restores the PAT, so you can run tests in the same R instance in which you develop.

Btw. with the keyring package, it will be even simpler, because the PATs can be in the keyring, and they won't get lost.

**Comment by jennybc** [comment]
Thanks! So I have my proposed set up working locally but will revise according to your suggestions. I think I also need to git config so that during tests I am committing as the dummy account. OTOH, if I don't, this could make me look **incredibly productive** on GitHub!

**Comment by gaborcsardi** [comment]
> I think I also need to git config so that during tests I am committing as the dummy account. 

Yeah, I guess so. But you can just set that up locally, without changing your global config.

> make me look incredibly productive on GitHub!

Nobody really takes that seriously any more: https://github.com/gelstudios/gitfiti

**Comment by jennybc** [comment]
Now that I think about it more, your approach seems like it will make things harder for me on Travis. I guess I could consult the env var that Travis sets to detect if tests are running there and override env vars (or not) accordingly.

**Comment by gaborcsardi** [comment]
No, I think it is all good, because the test PAT is not in the repo, so your "helper" will do nothing. (Smart helper!) But the env var will be set up by Travis, anyway. 

So I think the helper can just override whenever the `.env` exists in the current repo. Much easier than detecting Travis, and in the future you might want to use different CIs.

**Comment by jennybc** [comment]
OK wish me luck.
